perceval/archive.py

```diff
@@ -0,0 +1,312 @@
+
+        path = self.archive_folder_path
+        [os.remove(os.path.join(path, f)) for f in os.listdir(path)
+         if os.path.isfile(os.path.join(path, f)) and f.endswith(STORAGE_EXT)]
+
+    def archives(self, from_date=DEFAULT_DATETIME):
+        """List archives in folder path
+
+        :param from_date: date from when the archives can be retrieved
+
+        :return list of archive names
+        """
+
+        path = self.archive_folder_path
+        archives = [f.replace(STORAGE_EXT, "") for f in os.listdir(path)
+                    if os.path.isfile(os.path.join(path, f)) and f.endswith(STORAGE_EXT)]
+
+        return [os.path.join(path, f + STORAGE_EXT) for f in sorted(archives) if str_to_datetime(f) >= datetime_to_utc(from_date)]
+
+
+class Archive:
+    """Basic storage for Perceval.
+
+    This class allows to archive items for a
+    further recovery. Items are stored in a Sqlite db.
+
+    :param archive_path: path where to load/retrieve the archive
+    """
+
+    def __init__(self, archive_path):
+
+        self.archive_path = archive_path
+
+        self.db = sqlite3.connect(self.archive_path)
+
+    def __del__(self):
+
+        self.db.close()
+
+    @classmethod
+    def create(cls, archive_path):
+        """Create the DB table to store the archive data
+
+        :param archive_path: path where to create an archive
+        """
+        conn = sqlite3.connect(archive_path)
+
+        cursor = conn.cursor()
+        cursor.execute("DROP TABLE IF EXISTS archive")
+        cursor.execute("DROP TABLE IF EXISTS metadata")
+        conn.commit()
+
+        metadata_stmt = "CREATE TABLE metadata ( " \
+                        "origin TEXT, " \
+                        "backend_name TEXT, " \
+                        "backend_version TEXT, " \
+                        "from_date TEXT)"
+
+        archive_stmt = "CREATE TABLE archive(" \
+                       "id INTEGER PRIMARY KEY AUTOINCREMENT, " \
+                       "hash_request VARCHAR(256) UNIQUE NOT NULL, " \
+                       "url TEXT, " \
+                       "payload TEXT, " \
+                       "headers TEXT, " \
+                       "method TEXT, " \
+                       "stream TEXT, " \
+                       "verify TEXT, " \
+                       "response BLOB)"
+
+        cursor.execute(metadata_stmt)
+        cursor.execute(archive_stmt)
+        conn.commit()
+
+        cursor.close()
+        conn.close()
+
+    def from_date(self):
+        """Retrieve the from date data from the archive"""
+
+        cursor = self.db.cursor()
+        select_stmt = "SELECT from_date from metadata"
+        cursor.execute(select_stmt)
+        row = cursor.fetchone()
+        cursor.close()
+
+        if row:
+            found = pickle.loads(row[0])
+        else:
+            raise ArchiveError(cause="No metadata found")
+
+        return found
+
+    def metadata(self, origin, backend_name, backend_version, from_date):
```

Comment:
As I said above, I would name this method `init` or `initialize`, that will initialize metadata. This will be only called when the backend is fetching from a remote repository (so far, a HTTP or a Gerrit server).

Maybe we can have a property method named `metadata` to get these values from the database.